tests/test_breadth_first_search.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+from unittest.mock import Mock, call
+
+from src.breadth_first_search import breadth_first_search
+
+
+class TestBreadthFirstSearch(unittest.TestCase):
+    def setUp(self):
+        self.fixture = {
+            "first": ["second", "third"],
+            "second": ["third", "fourth", "me-1", "sixth"],
+            "third": ["first", "fifth"],
+            "fourth": ["me-2"],
+            "fifth": ["second", "seventh"],
+            "sixth": [],
+            "seventh": [],
+            "me-1": [],
+            "me-2": []
+        }
+
+    def test_empty(self):
+        self.assertEqual(
+            breadth_first_search(
+                {},
+                "first",
+                lambda x: x == "second"
+            ),
+            False
+        )
+
+    def test_find(self):
+        self.assertEqual(
+            breadth_first_search(
+                self.fixture,
+                "first",
+                lambda x: x == "seventh"
+            ),
+            True
+        )
+
+    def test_not_find(self):
+        self.assertEqual(
+            breadth_first_search(
+                self.fixture,
+                "fourth",
+                lambda x: x == "fifth"
+            ),
+            False
+        )
+
+    def test_not_find_when_absent(self):
+        self.assertEqual(
+            breadth_first_search(
+                self.fixture,
+                "first",
+                lambda x: x == "no"
+            ),
+            False
+        )
+
+    def test_is_bread_first(self):
+        spy = Mock(wraps=lambda x: x.startswith("me"))
+        result = breadth_first_search(self.fixture, "first", spy)
+
+        self.assertEqual(result, True)
+        self.assertEqual(spy.mock_calls, [
+            call("second"),
+            call("third"),
+            call("fourth"),
+            call("me-1")
```

Comment:
`me-1` will be found with depth-first search too and lambda calls will be same =)